### PR TITLE
Introduce purs-tidy formatter

### PR DESCRIPTION
**Pull request by thomashoneyman**
**Description of the change**
Introduces the [`purs-tidy`](https://github.com/natefaubion/purescript-tidy) formatter. This formatter is a lightly opinionated tool we use to maintain a consistent style in the contrib libraries.

We ordinarily restrict to tools provided by the `core` or `contrib` projects. This tool is not, but it was developed and is maintained by core team members, and because it's a formatter it can't block maintenance or release of this library even in the event something goes catastrophically wrong with it.

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation in the README and/or documentation directory
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by thomashoneyman** [comment]
@purescript-contrib/maintainers This is a draft pull request demonstrating introducing the `purs-tidy` formatter. This has been opened by an automated script, and if it looks good then all we really need to do is evaluate formatting decisions made by `purs-tidy` to fix up anything particularly wonky before committing.

If the formatting looks fine then please go ahead and approve, and if it doesn't then please comment on the PR with a suggested fix. I'll do the pull requests in batches and will leave each batch open for a few days for review feedback before merging them in.

If you would like to _not_ be assigned or pinged when new batches go up, please let me know and I'll remove you!

File: src/Ace/KeyBinding.purs [hunk 1]
```diff
@@ -10,7 +10,8 @@ import Partial.Unsafe (unsafePartial)
 import Web.UIEvent.KeyboardEvent (KeyboardEvent)
 
 type KeyboardHandler
-  = { editor :: Editor }
+  =
+  { editor :: Editor }
```
Comment (JordanMartinez) [inline src/Ace/KeyBinding.purs:13]: The `=` here should be on the same line as `KeyboardHandler`, right?

File: src/Ace/Types.purs [hunk 2]
```diff
@@ -79,10 +79,10 @@ readPosition e = do
   pure $ Position { row, column }
 
 getRow :: Position -> Int
-getRow (Position {row : row}) = row
+getRow (Position { row: row }) = row
```
Comment (JordanMartinez) [inline src/Ace/Types.purs]: Should this just be `(Position { row }) = row`

File: src/Ace/Types.purs [hunk 3]
```diff
@@ -79,10 +79,10 @@ readPosition e = do
...
 
 getColumn :: Position -> Int
-getColumn (Position {column : column}) = column
+getColumn (Position { column: column }) = column
```
Comment (JordanMartinez) [inline src/Ace/Types.purs]: Should this just be `(Position { column }) = column`

**Review by JordanMartinez — COMMENTED** [review]
A few comments, but otherwise LGTM.

**Pull request merged**